app/models/manageiq/providers/redhat/inventory/collector/target_collection.rb

```diff
@@ -113,22 +113,30 @@ def vms
     v
   end
 
+  def select_vms(references_lst)
+    references_lst.select { |ref| ref.include?('vms') }
+  end
+
+  def select_templates(references_lst)
+    references_lst.select { |ref| ref.include?('templates') }
```

Comment:
both select_* methods differ by a string. Can you accept it as param and have only one?

Reply:
I looked at it, and although it is a little code duplication it looks better to me like this. Because passing a parameter for this does not look so good, the way would be to do method_missing and then we should do responds_to also etc.
But if you will insist I will change it.